### ISSUE TITLE
Proposal: list-database-roles

### ISSUE THREAD

**Issue by Symbolics** [marijnh/Postmodern#207]
We currently have a list-database-users function, but none for listing _roles_. This proposal is to create a function for listing the roles in the current database.

**Comment by sabracrolleton** [comment]
These days roles and users are almost the same in postgresql. The only difference between the two functions would be that list-database-roles would also show entities that cannot log in. Are you looking for just a list of the roles or something more like the output of /du?

**Comment by Symbolics** [comment]
Mostly looking something like the output of /du.

**Comment by sabracrolleton** [comment]
the sql for /du is

```
SELECT r.rolname, r.rolsuper, r.rolinherit,
  r.rolcreaterole, r.rolcreatedb, r.rolcanlogin,
  r.rolconnlimit, r.rolvaliduntil,
  ARRAY(SELECT b.rolname
        FROM pg_catalog.pg_auth_members m
        JOIN pg_catalog.pg_roles b ON (m.roleid = b.oid)
        WHERE m.member = r.oid) as memberof
, r.rolreplication
, r.rolbypassrls
FROM pg_catalog.pg_roles r
WHERE r.rolname !~ '^pg_'
ORDER BY 1;
```

so, basically returning all those items?

**Comment by Symbolics** [comment]
Essentially, yes.

    On Saturday, September 14, 2019, 9:39:00 PM GMT+8, Sabra Crolleton <notifications@github.com> wrote:  
 
 
the sql for /du is
SELECT r.rolname, r.rolsuper, r.rolinherit,
  r.rolcreaterole, r.rolcreatedb, r.rolcanlogin,
  r.rolconnlimit, r.rolvaliduntil,
  ARRAY(SELECT b.rolname
        FROM pg_catalog.pg_auth_members m
        JOIN pg_catalog.pg_roles b ON (m.roleid = b.oid)
        WHERE m.member = r.oid) as memberof
, r.rolreplication
, r.rolbypassrls
FROM pg_catalog.pg_roles r
WHERE r.rolname !~ '^pg_'
ORDER BY 1;

so, basically returning all those items?

**Comment by sabracrolleton** [comment]
Will add

**Comment by sabracrolleton** [comment]
Done in today's commit. Sorry it took so long.
